### PR TITLE
Add lll linter

### PR DESCRIPTION
**Pull request by eqrx**
Enable the golangci-lint `lll` linters which ensures that code lines may not be longer that 120 chars (non whitespace)

### CODE REVIEW

File: internal/controllers/hostedclusters/hostedcluster_controller.go [hunk 1]
```diff
@@ -144,8 +145,7 @@ func (c *HostedClusterController) desiredPackage(cluster *v1beta1.HostedCluster)
 	return pkg, nil
 }
 
-// From
-// https://github.com/openshift/hypershift/blob/9c3e998b0b37bedce07163a197e0bf30339e627e/hypershift-operator/controllers/manifests/manifests.go#L13
+// From https://github.com/openshift/hypershift/blob/v0.1.13/hypershift-operator/controllers/manifests/manifests.go#L16
```
Comment (erdii) [inline internal/controllers/hostedclusters/hostedcluster_controller.go:148]: Nit: Tags can move. Commits hashes can't.

File: internal/controllers/objectdeployments/adapter_test.go [hunk 2]
```diff
@@ -98,7 +98,10 @@ func TestGenericObjectSet_GetObjects(t *testing.T) {
 			},
 				nil,
 			),
-			expectedObjectIDs: []string{"/ConfigMap/namespace1/cm1", "/ConfigMap/namespace2/cm2", "apps/Deployment/default/deployment1", "/Secret/default/secret1"},
+			expectedObjectIDs: []string{
+				"/ConfigMap/namespace1/cm1", "/ConfigMap/namespace2/cm2",
+				"apps/Deployment/default/deployment1", "/Secret/default/secret1",
+			},
```
Comment (erdii) [inline internal/controllers/objectdeployments/adapter_test.go:104]: Why not 

```suggestion
			expectedObjectIDs: []string{
				"/ConfigMap/namespace1/cm1",
				"/ConfigMap/namespace2/cm2",
				"apps/Deployment/default/deployment1",
				"/Secret/default/secret1",
			},
```

?

To me that is way easier to read when glancing over and searching for "that one string"

File: internal/controllers/objectdeployments/adapter_test.go [hunk 3]
```diff
@@ -113,7 +116,10 @@ func TestGenericObjectSet_GetObjects(t *testing.T) {
 			},
 				nil,
 			),
-			expectedObjectIDs: []string{"/ConfigMap/default-1/cm1", "/ConfigMap/default-1/cm2", "apps/Deployment/default-1/deployment1", "/Secret/default-1/secret1"},
+			expectedObjectIDs: []string{
+				"/ConfigMap/default-1/cm1", "/ConfigMap/default-1/cm2",
+				"apps/Deployment/default-1/deployment1", "/Secret/default-1/secret1",
+			},
```
Comment (erdii) [inline internal/controllers/objectdeployments/adapter_test.go:122]: :bookmark:

File: internal/controllers/objectsetphases/objectsetphase_controller.go [hunk 4]
```diff
@@ -305,6 +307,7 @@ func (c *GenericObjectSetPhaseController) SetupWithManager(
 
 	return ctrl.NewControllerManagedBy(mgr).
 		For(objectSetPhase).
-		WatchesRawSource(c.dynamicCache.Source(), c.ownerStrategy.EnqueueRequestForOwner(objectSetPhase, mgr.GetRESTMapper(), false)).
-		Complete(c)
+		WatchesRawSource(c.dynamicCache.Source(),
+			c.ownerStrategy.EnqueueRequestForOwner(objectSetPhase, mgr.GetRESTMapper(), false),
+		).Complete(c)
```
Comment (erdii) [inline internal/controllers/objectsetphases/objectsetphase_controller.go:312]: What do you think of
```suggestion
		WatchesRawSource(
			c.dynamicCache.Source(),
			c.ownerStrategy.EnqueueRequestForOwner(objectSetPhase, mgr.GetRESTMapper(), false),
		).Complete(c)
```

**Comment by erdii** [comment]
I stopped reviewing this PR halfway through because this is essentially 2 PRs in one and each of them touches so many lines that I had a hard time keeping track of everything:

1. adding lll and fixing line length accordingly
2. replacing all instances of `inferface{}` with `any`

Can you please split them into separate PRs? :pray:

**Review by erdii — CHANGES_REQUESTED** [review]
^

**Comment by eqrx** [comment]
Split the PRs apart

**Pull request merged**